### PR TITLE
FOGL-1440 - Update Configuration Manager and its tests to use async storage client - A

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
This PR updates core configuration files, code + tests, only. Other client code such Scheduler etc will be updated separately. Currently, tests of client code are failing as under:

```
tests/unit/python/foglamp/common/test_audit_logger.py ..........                                                                          [  0%]
tests/unit/python/foglamp/common/test_configuration_manager.py ........................................................................   [  6%]
tests/unit/python/foglamp/common/test_jqfilter.py ....                                                                                    [  6%]
tests/unit/python/foglamp/common/test_logger.py .....                                                                                     [  7%]
tests/unit/python/foglamp/common/test_process.py ........                                                                                 [  7%]
tests/unit/python/foglamp/common/test_service_record.py .......                                                                           [  8%]
tests/unit/python/foglamp/common/test_statistics.py ..................                                                                    [  9%]
tests/unit/python/foglamp/common/microservice_management_client/test_microservice_management_client.py .................................. [ 12%]
.....                                                                                                                                     [ 12%]
tests/unit/python/foglamp/common/storage_client/test_payload_builder.py ................................................................. [ 18%]
.s..............                                                                                                                          [ 19%]
tests/unit/python/foglamp/common/storage_client/test_sc_exceptions.py ..........                                                          [ 20%]
tests/unit/python/foglamp/common/storage_client/test_storage_client.py ..............................                                     [ 22%]
tests/unit/python/foglamp/common/storage_client/test_utils.py ..................                                                          [ 24%]
tests/unit/python/foglamp/common/web/test_middleware.py ......                                                                            [ 24%]
tests/unit/python/foglamp/plugins/common/test_plugins_common_utils.py ....                                                                [ 24%]
tests/unit/python/foglamp/plugins/north/common/test_common.py ..................................................                          [ 28%]
tests/unit/python/foglamp/plugins/north/http_north/test_http_north.py .ss....                                                             [ 29%]
tests/unit/python/foglamp/plugins/north/ocs/test_ocs.py ............                                                                      [ 30%]
tests/unit/python/foglamp/plugins/north/omf/test_omf.py ......................................                                            [ 33%]
tests/unit/python/foglamp/plugins/south/cc2650async/test_cc2650async.py ........                                                          [ 34%]
tests/unit/python/foglamp/plugins/south/cc2650poll/test_cc2650poll.py ...........                                                         [ 34%]
tests/unit/python/foglamp/plugins/south/coap_listen/test_coap_listen.py ...........                                                       [ 35%]
tests/unit/python/foglamp/plugins/south/common/test_sensortag_cc2650.py ........................s                                         [ 37%]
tests/unit/python/foglamp/plugins/south/http_south/test_http_south.py ...............                                                     [ 38%]
tests/unit/python/foglamp/services/common/test_avahi.py ......                                                                            [ 39%]
tests/unit/python/foglamp/services/common/test_microservice.py ....                                                                       [ 39%]
tests/unit/python/foglamp/services/common/test_services_common_utils.py ....                                                              [ 40%]
tests/unit/python/foglamp/services/common/microservice_management/test_instance.py ..........                                             [ 40%]
tests/unit/python/foglamp/services/core/test_connect.py ...                                                                               [ 41%]
tests/unit/python/foglamp/services/core/test_main.py .                                                                                    [ 41%]
tests/unit/python/foglamp/services/core/test_server.py sssssssssssssssss..sssss.......FFFFFFFFF.FFF.FF......................              [ 46%]
tests/unit/python/foglamp/services/core/test_user_model.py ...........................FFFFFF.....s...........                             [ 50%]
tests/unit/python/foglamp/services/core/api/test_audit.py ..........................                                                      [ 52%]
tests/unit/python/foglamp/services/core/api/test_auth_mandatory.py ..............................................................         [ 57%]
tests/unit/python/foglamp/services/core/api/test_auth_optional.py ......................................                                  [ 60%]
tests/unit/python/foglamp/services/core/api/test_backup_restore.py ......................................                                 [ 63%]
tests/unit/python/foglamp/services/core/api/test_browser_assets.py ..........................................                             [ 67%]
tests/unit/python/foglamp/services/core/api/test_certificate_store.py ................FF                                                  [ 68%]
tests/unit/python/foglamp/services/core/api/test_common_ping.py FFFFFFFF.                                                                 [ 69%]
tests/unit/python/foglamp/services/core/api/test_configuration.py ....................................                                    [ 72%]
tests/unit/python/foglamp/services/core/api/test_scheduler_api.py ....................................................................... [ 77%]
.......                                                                                                                                   [ 78%]
tests/unit/python/foglamp/services/core/api/test_service.py .........s                                                                    [ 79%]
tests/unit/python/foglamp/services/core/api/test_statistics_api.py ...............................                                        [ 81%]
tests/unit/python/foglamp/services/core/api/test_support.py ................                                                              [ 82%]
tests/unit/python/foglamp/services/core/interest_registry/test_change_callback.py .....                                                   [ 83%]
tests/unit/python/foglamp/services/core/interest_registry/test_interest_registry.py ........                                              [ 83%]
tests/unit/python/foglamp/services/core/scheduler/test_scheduler.py ......s..............s.............s..............s......s            [ 88%]
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py .........                                                    [ 89%]
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_exceptions.py ......                                                     [ 89%]
tests/unit/python/foglamp/services/core/service_registry/test_exceptions.py .....                                                         [ 90%]
tests/unit/python/foglamp/services/core/service_registry/test_monitor.py .s                                                               [ 90%]
tests/unit/python/foglamp/services/core/service_registry/test_service_registry.py .........                                               [ 90%]
tests/unit/python/foglamp/services/south/test_ingest.py ....ss........                                                                    [ 92%]
tests/unit/python/foglamp/services/south/test_services_south_main.py .                                                                    [ 92%]
tests/unit/python/foglamp/services/south/test_services_south_server.py ...............                                                    [ 93%]
tests/unit/python/foglamp/tasks/north/test_sending_process.py ...........................................................                 [ 98%]
tests/unit/python/foglamp/tasks/purge/test_purge.py ................                                                                      [ 99%]
tests/unit/python/foglamp/tasks/purge/test_purge_main.py .                                                                                [ 99%]
tests/unit/python/foglamp/tasks/statistics/test_statistics_history.py ......                                                              [ 99%]
tests/unit/python/foglamp/tasks/statistics/test_statistics_main.py .                                                                      [100%]
```

### CODE REVIEW

File: python/foglamp/common/configuration_manager.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def __init__(self, storage=None):
             await audit.information('CONAD', {'name': category_name, 'category': category_val})
             payload = PayloadBuilder().INSERT(key=category_name, description=category_description,
                                               value=category_val).payload()
-            result = self._storage.insert_into_tbl("configuration", payload)
+            result = await self._storage.insert_into_tbl("configuration", payload)
             response = result['response']
         except KeyError:
```
Comment (praveen-garg) [inline python/foglamp/common/configuration_manager.py]: all such except blocks needs to be chnaged to `expect StorageServerError`

File: python/foglamp/common/configuration_manager.py [hunk 2]
```diff
@@ -158,17 +158,27 @@ def __init__(self, storage=None):
...
             raise ValueError(result['message'])
+        except StorageServerError as ex:
+            err_response = ex.error
+            # if key error in next, it will be automatically in parent except block
```
Comment (praveen-garg) [inline python/foglamp/common/configuration_manager.py]: which one?

Comment (amarendra-dianomic) [reply python/foglamp/common/configuration_manager.py]: In the calling client code.

**Comment by amarendra-dianomic** [comment]
Due to highly integrated code, it is not possible to modify and test code, for converting from sync storage client to async storage client, related to FOGL-1394, FOGL-1395, FOGL-1396, FOGL-1397, FOGL-1401, FOGL-1402, FOGL-1403.  Hence closing PRs #857, #858 and #859 in favour of this PR #874.

**Pull request closed without merging**